### PR TITLE
Add buttons for Up and New File in the load/save dialog and fix #2233

### LINKED ISSUE #2233: Don't show (new file) option in save dialog when displaying Windows drive letters
On Windows, when clicking "(up)" as far as you go in the save dialog, you get a list of the drive letters. Since this is not a folder, and you can't save anything here, there's no reason to have a "(new file)" option in the list that does nothing.
openrct2_new_file

### PR DESCRIPTION
**Pull request by osconnoisseur**
I've changed loadsave.c to fix #2233 and show the "(new)" item only if the path is not an empty string. The path is an empty string when the drive letters are listed. As usual, this needs proper testing. Also I don't know what the real purpose of the conditional statement on line 649 is. It was preventing the "(new)" item from showing up on the root of a drive, which should be possible, so I removed the offending line.

### CODE REVIEW

**Comment by Broxzier** [comment]
You'll probably need to check `platform_get_drives()` before setting `includeNewItem` to false.

**Comment by janisozaur** [comment]
I wonder if this code work on Linux, I have never really tried going all the way `(up)`.

Also: usually, directory up is indicated by `..` on all platforms, so `(up)` (a valid file name) is slightly misleading.

**Comment by Gymnasiast** [comment]
It would be better to move (new file) and (up) out of the list and into separate buttons. That will be a lot clearer.

**Comment by janisozaur** [comment]
I like that idea @Gymnasiast

**Comment by IntelOrca** [comment]
Or change the colour of those items.

**Comment by Gymnasiast** [comment]
That wouldn't be as clear, and making separate button isn't that much extra work. I always try to think from the average user, not from our own level.

**Comment by osconnoisseur** [comment]
Windows is the only major operating system that has drive letters, so I'm for just separating the platform_get_drive_letters() and includeNewItem from the unix code.

AmigaOS, AROS, and Morph OS have device names, which are similar, but platform_get_drive_letters() wouldn't work for that.

**Comment by janisozaur** [comment]
I just checked, when going all the way `(up)` in Linux, you get an empty list with `(new file)` only.
zaznaczenie_045
Can it possibly be fixed with this commit as well?

**Comment by osconnoisseur** [comment]
@janisozaur, I'm working on that right now.

Can someone explain the purpose of the seemingly pointless 'browsable' parameter of window_loadsave_populate_list()? It's set to true every time the function is called. When would the list not be browsable?

**Comment by janisozaur** [comment]
https://github.com/OpenRCT2/OpenRCT2/commit/10babd3c?w=1 this is what introduced the option

**Comment by janisozaur** [comment]
github says your commit is non-mergeable, can you rebase it on top of more recent revision to fix conflicts?

File: src/windows/loadsave.c [hunk 1]
```diff
@@ -537,7 +552,7 @@ static void window_loadsave_scrollpaint(rct_window *w, rct_drawpixelinfo *dpi, i
 	rct_string_id stringId, templateStringId = 3165;
 	char *templateString;
 
-	gfx_fill_rect(dpi, dpi->x, dpi->y, dpi->x + dpi->width - 1, dpi->y + dpi->height - 1, ColourMapA[w->colours[1]].mid_light);
+	gfx_fill_rect(dpi, dpi->x, dpi->y, dpi->x + dpi->width - 1, dpi->y + dpi->height - 1, RCT2_ADDRESS(0x0141FC48,uint8)[w->colours[1] * 8]);
```
Comment (IntelOrca) [inline src/windows/loadsave.c]: merge error here

File: src/windows/loadsave.c [hunk 2]
```diff
@@ -153,18 +156,18 @@ rct_window *window_loadsave_open(int type, char *defaultName)
...
-	_defaultName[0] = 0;
-
-	if (!str_is_null_or_empty(defaultName)) {
-		safe_strncpy(_defaultName, path_get_filename(defaultName), sizeof(_defaultName));
-		path_remove_extension(_defaultName);
+	_defaultName[0] = 0;  
+	
+	if (!str_is_null_or_empty(defaultName)) {  
+		safe_strncpy(_defaultName, path_get_filename(defaultName), sizeof(_defaultName));  
+		path_remove_extension(_defaultName);
```
Comment (IntelOrca) [inline src/windows/loadsave.c]: You appear to be adding whitespace to the lines above, remove these.

File: src/windows/loadsave.c [hunk 3]
```diff
@@ -795,7 +783,7 @@ static void window_loadsave_select(rct_window *w, const char *path)
 			SDL_RWclose(rw);
 			if (success) {
 
-				safe_strncpy(gScenarioSavePath, path, MAX_PATH);
+				safe_strncpy(gScenarioSavePath, path, MAX_PATH);
```
Comment (IntelOrca) [inline src/windows/loadsave.c]: remove whitespace at the end of this line

File: src/windows/loadsave.c [hunk 4]
```diff
@@ -537,7 +552,7 @@ static void window_loadsave_scrollpaint(rct_window *w, rct_drawpixelinfo *dpi, i
 	rct_string_id stringId, templateStringId = 3165;
 	char *templateString;
 
-	gfx_fill_rect(dpi, dpi->x, dpi->y, dpi->x + dpi->width - 1, dpi->y + dpi->height - 1, ColourMapA[w->colours[1]].mid_light);
+	gfx_fill_rect(dpi, dpi->x, dpi->y, dpi->x + dpi->width - 1, dpi->y + dpi->height - 1, RCT2_ADDRESS(0x0141FC48,uint8)[w->colours[1] * 8]);
```
Comment (IntelOrca) [inline src/windows/loadsave.c]: merge error here

**Comment by osconnoisseur** [comment]
Can someone who is better with git than I am verify that this is finally mergeable? I had a hell of a time trying to resolve revision conflicts with this PR.

**Comment by Gymnasiast** [comment]
Considering there is 1 commit and Github considers it mergeable, it is perfect Git-wise. Can't say much about the actual code though.

**Comment by duncanspumpkin** [comment]
Can you change the english_uk entries to not use the brackets and sentence case the words. They look out of place.

**Comment by osconnoisseur** [comment]
@duncanspumpkin, I did that for all languages, if that's okay.

**Comment by Gymnasiast** [comment]
No, that is not okay. Other languages are maintained at OpenRCT2/Localisation.

**Pull request merged**